daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/AstRewriter.scala

```diff
@@ -52,6 +52,7 @@ private[digitalasset] class AstRewriter(
     if (typeRule.isDefinedAt(x)) typeRule(x)
     else
       x match {
+        case TSynApp(_, _) => throw new RuntimeException("TODO #3616,AstRewriter,TSynApp")
```

Comment:
The `AstRewriter` is only used in `EncodeV1` so those two should be fixed together and as mentioned above, we can leave that for a separate PR.

Reply:
ok